File: libs/modules/gc/src/commonMain/kotlin/com/smellouk/konitor/gc/GcInfo.kt
package com.smellouk.konitor.gc

import com.smellouk.konitor.api.Info

data class GcInfo(
    val gcCount: Long,
    val gcPauseMs: Long,
    val gcCountDelta: Long,
    val gcPauseMsDelta: Long
) : Info {
    companion object {
        val INVALID = GcInfo(-1L, -1L, -1L, -1L)
        val UNSUPPORTED = GcInfo(-2L, -2L, -2L, -2L)
    }
}

File: libs/ui/flutter/lib/src/models/gc_info.dart
class GcInfo {
  final double gcCountDelta;
  final double gcPauseMsDelta;
  final double gcCount;

  const GcInfo({
    required this.gcCountDelta,
    required this.gcPauseMsDelta,
    required this.gcCount,
  });

  factory GcInfo.fromMap(Map<dynamic, dynamic> m) => GcInfo(
        gcCountDelta: (m['gcCountDelta'] as num).toDouble(),
        gcPauseMsDelta: (m['gcPauseMsDelta'] as num).toDouble(),
        gcCount: (m['gcCount'] as num).toDouble(),
      );
}
